### DISCUSSION TITLE
Does 'startRecordingScreen and stopScreenRecording commands works for emulators with all api levels

### DISCUSSION THREAD

**Discussion by sachinvermacs** [appium/appium#15644] [Q&A] [answered]
Does Appium support screen recording commands for emulators?
I tried using both driver.startRecordingScreen(); and driver.stopRecordingScreen(); but behavior is different across the API levels.

**Comment by KazuCocoa** [comment, marked as answer]
Appium calls adb command in the backend for Android. The behavior depends on how Android devices/emulators works.
https://github.com/appium/appium-android-driver/blob/0e3d9b31842fd68b1497c94e9ddd1e58ff0de9b9/lib/commands/recordscreen.js#L53
https://github.com/appium/appium-adb/blob/e3108032e1cbfb298bdc5606f7647112084a0982/lib/tools/adb-commands.js#L1595-L1624

**Reply by sachinvermacs** [reply]
Thanks, will check.

**Reply by sachinvermacs** [reply]
I tried with this getAndroidDriver().startRecordingScreen(new AndroidStartScreenRecordingOptions().withVideoSize("1280x720")
					.enableForcedRestart().withTimeLimit(Duration.ofSeconds(300)));
and it worked.
